### DISCUSSION TITLE
All yearly bills are wrongly expected for the year after

### DISCUSSION THREAD

**Discussion by ryester19** [firefly-iii/firefly-iii#8080] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

I have this bill, which should be expected for the period of October 16, 2023 - October 22, 2023:

Date: 10/22/2023
Repeats: yearly

On the Bills page, I see this bill as having a next expected match of October 22, 2024, and that it's "not expected this period". However, on the dashboard, I can see the amount was included in the "Bills to pay" card. 

This cosmetic bug appears to be affecting all my yearly bills, but not any of my monthly ones

### Debug information

Debug information generated at 2023-10-17 10:27:31 for Firefly III version **v6.0.27**.

<table><thead><tr><th colspan="2">System information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Firefly III</td><td>v6.0.27 / v2.0.10 / 20 (exp. 20)</td></tr><tr><td>PHP version</td><td>8.2.11 / apache2handler / Linux</td></tr><tr><td>BCscale</td><td>12</td></tr><tr><td>Error reporting</td><td>Display: Off, reporting: ALL errors</td></tr><tr><td>Max upload</td><td>67108864 (64 MB)</td></tr><tr><td>Database drivers</td><td>mysql, <strong>*pgsql*</strong>, sqlite, </td></tr><tr><td>Docker build</td><td><span>#</span>723, base <span>#</span>39</td></tr></tbody></table><table><thead><tr><th colspan="2">Firefly III information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Timezone</td><td>America/New_York + America/New_York</td></tr><tr><td>App environment</td><td>local, debug: true</td></tr><tr><td>Layout</td><td>v1</td></tr><tr><td>Logging</td><td>debug, stack / (empty)</td></tr><tr><td>Cache driver</td><td>file</td></tr><tr><td>Default language and locale</td><td>en_US + equal</td></tr><tr><td>Trusted proxies</td><td>**</td></tr><tr><td>Login provider & user guard</td><td>eloquent / web</td></tr><tr><td>Login headers</td><td>N/A + N/A</td></tr><tr><td>Stateful domains</td><td></td></tr><tr><td>Last cron job</td><td>2023-10-17 09:32:11 (55 minutes ago)</td></tr><tr><td>Mailer</td><td>smtp</td></tr></tbody></table><table><thead><tr><th colspan="2">User-specific information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>User</td><td><span>#</span>1 of 1</td></tr><tr><td>User flags</td><td><span title="Has piggy banks">:pig:</span> <span title="Has rules">:wrench:</span> <span title="Has object groups">:bookmark_tabs:</span> <span title="Has subscriptions">:email:</span></td></tr><tr><td>Session start</td><td>2023-10-16 00:00:00</td></tr><tr><td>Session end</td><td>2023-10-22 23:59:59</td></tr><tr><td>View range</td><td>1W</td></tr><tr><td>User language</td><td>en_US</td></tr><tr><td>User locale</td><td>en_US</td></tr><tr><td>Locale(s) supported</td><td>en_US.utf8: :white_check_mark:<br>en_US.UTF-8: :white_check_mark:<br></td></tr><tr><td>User agent</td><td>Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/118.0</td></tr></tbody></table>

**Comment by JC5** [comment]
> Date: 10/22/2023

The bill has not started its cycle yet. Change the date to 10/22/2022.
